designs/Net5PlatformAndPlatformVersion.md

```diff
@@ -0,0 +1,177 @@
+# net5.0 TFM Platform and PlatformVersion implementation
+
+This document is meant to describe the various parts of the code that need to be updated in order to support platforms in net5.0 target framework identifiers. That is, the optional part after a `-` following `net5.0`. For example, `net5.0-ios14.1`. Note that this ONLY covers the platform part, because [initial support for `net5.0` has already been added](https://github.com/NuGet/NuGet.Client/pull/3265).
+
+For more details on the feature itself, please refer to [the full spec](https://github.com/terrajobst/designs/blob/master/accepted/2020/net5/net5.md).
+
+## class NuGetFramework
+
+Source: src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs
+
+### Fields
+
+Three additional fields will be added:
+
+```csharp
+        /// <summary>
+        /// Framework Platform (net5.0+)
+        /// </summary>
+        public string Platform
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Framework Platform Version (net5.0+)
+        /// </summary>
+        public Version PlatformVersion
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// True if platform is non-empty (net5.0+)
+        /// </summary>
+        public bool HasPlatform

```

Comment:
I know other properties in the class have `HasWhatever` properties, but I don't like the pattern. Where is it used? Why doesn't the caller do `!string.IsNullOrEmpty(PlatformIdentifier)` instead?

Reply:
I think `!string.IsNullOrEmpty()` isn't as descriptive as having a boolean to check against that knows the semantics of the value being missing. That said, I'll remove this for now, because I don't know where we'll use it -- I might bring it back if I find a case where it's actually needed for public API purposes.